src/core/Akka.Cluster/Cluster.cs

```diff
@@ -78,6 +79,9 @@ public Cluster(ActorSystemImpl system)
 
             _scheduler = CreateScheduler(system);
 
+            // it has to be lazy - otherwise if downing provider will init a cluster itself, it will deadlock
+            _downingProvider = new Lazy<IDowningProvider>(() => Akka.Cluster.DowningProvider.Load(Settings.DowningProviderType, system), LazyThreadSafetyMode.ExecutionAndPublication);
```

Comment:
Does this also need to be thread-safe for execution? Or does it only need to be thread-safe upon creation?